make_site.py

```diff
@@ -455,12 +457,48 @@ class User:
     fullname: str
     lon: float
     lat: float
-    zulip: str
     github: Optional[str] = None
     website: Optional[str] = None
 
-with (DATA/'users.yaml').open('r', encoding='utf-8') as h_file:
-    users = [User(**e) for e in yaml.safe_load(h_file)]
+client = zulip.Client(
+    email='map-scraper-bot@leanprover.zulipchat.com', 
+    site='https://leanprover.zulipchat.com', 
+    api_key=os.environ.get('ZULIP_KEY')) if 'ZULIP_KEY' in os.environ else None
+
+# Zulip custom profile fields are tracked by ID, not by name
+profile_data_fields = {
+    'website': '5150',
+    'latitude': '5148',
+    'longitude': '5149',
+    'github': '3658'
+}
+
+def get_user_field(user, field):
+    try:
+        return user['profile_data'][profile_data_fields[field]]['value']
+    except KeyError:
+        return None
+
+def validate_coordinates(user):
+    try:
+        float(get_user_field(user, 'latitude'))
+        float(get_user_field(user, 'longitude'))
+        return True
+    except Exception:
+        return False
+
+
+zulip_users = [user for user in client.get_members({"include_custom_profile_fields": True})['members'] 
+               if not user['is_bot'] and user['is_active'] and validate_coordinates(user)] if 'ZULIP_KEY' in os.environ else []
+
+users = [
+    User(
+        fullname=user['full_name'], 
+        lon=float(get_user_field(user, 'longitude')), 
+        lat=float(get_user_field(user, 'latitude')), 
+        github=get_user_field(user, 'github'),
+        website=get_user_field(user, 'website')) 
+    for user in zulip_users]
```

Comment:
Perhaps:
```suggestion
def get_users():
  if client is None:
    return

  for user in client.get_members({"include_custom_profile_fields": True})['members']:
    if user['is_bot'] or not user['is_active']:
      continue
    try:
      lat = float(get_user_field(user, 'latitude'))
      lon = float(get_user_field(user, 'longitude'))
    except Exception:
      continue
    yield User(
        fullname=user['full_name'], 
        lon=float(get_user_field(user, 'longitude')), 
        lat=float(get_user_field(user, 'latitude')), 
        github=get_user_field(user, 'github'),
        website=get_user_field(user, 'website'))
  
users = list(get_users())
```
to avoid getting the coordinates twice.

Reply:
Because I don't know how many times we iterate over it below, and I figured this was  least likely to break anything.
Indeed, if we're careful we could leave it as an iterator.